earthpy/mask.py

```diff
@@ -86,105 +86,107 @@
+        mask_arr[temp_mask] = 1
+        mask_arr[~temp_mask] = 0
 
-    # Mask the values
-    mask_arr[temp_mask] = 1
-    mask_arr[~temp_mask] = 0
+        return mask_arr
 
-    return mask_arr
+    else:
+        raise ValueError(
+            """List of values provided for the mask does 
+                             not exist in your mask array."""
+        )
 
 
-def apply_cloud_mask(arr, the_mask):
-    """Take an input raster numpy array in band order (bands, rows,cols)
-     and apply a mask to the array
+def _apply_mask(arr, input_mask):
+    """Applies a single dimension mask to the provided array.
 
     Parameters
     -----------
     arr : numpy array
-        An array in rasterio (band, row, col) order
+        The original numpy array in rasterio (band, row, col) order
+        that needs a mask applied.
 
-    the_mask : single dimension (band) numpy array
-        A numpy array containing a qa values for raster pixels that should be
-        masked - ex landsat pixel_qa layer.
+    input_mask : numpy array
+        A numpy array containing O's and 1's where the 1's indicate where the
+        mask is applied.
 
     Returns
     -----------
-    masked arr : a masked numpy array
-        A masked numpy array with the mask having the same dimensions as arr
+    numpy array
+        The original numpy array with the mask applied to cover up issue pixels.
     """
 
-    # Test if the_mask is numpy array w values == 1 for masked
-    if not np.any(the_mask == 1):
+    # Test if input_mask is numpy array w values == 1 for masked
+    if not np.any(input_mask == 1):
         raise ValueError("Mask requires values of 1 (True) to be applied.")
 
-    try:
-        # Create a mask for all bands in the landsat scene
-        cl_mask = np.broadcast_to(the_mask == 1, arr.shape)
-    except AttributeError:
-        raise AttributeError("Input arr should be a numpy array")
+    cover_mask = np.broadcast_to(input_mask == 1, arr.shape)
 
     # If the user provides a masked array, combine masks
     if isinstance(arr, np.ma.MaskedArray):
-        cl_mask = np.logical_or(arr.mask, cl_mask)
+        cover_mask = np.logical_or(arr.mask, cover_mask)
 
     # Return combined mask
-    return ma.masked_array(arr, mask=cl_mask)
+    return ma.masked_array(arr, mask=cover_mask)
 
 
-def make_apply_mask(arr, mask_arr, vals):
+def mask_pixels(arr, mask_arr, vals=None):
     """Take an input array to be masked, single band mask layer such as a
-    pixel_qa layer for MODIS or Landsat and apply a mask given a range of
-    vals to mask.
+    pixel_qa layer for MODIS or Landsat, and apply a mask given a range of
+    values to mask. This function can also be passed a previously created
+    mask in place of a qa layer and a list of values.
 
     Parameters
     -----------
     arr : numpy array
-        An array in rasterio (band, row, col) order
+        The desired array to mask in rasterio (band, row, col) order.
```

Comment:
```suggestion
        The array to mask in rasterio (band, row, col) order.
```

Reply:
Super minor suggestion here - but "desired" array might be confusing

Reply:
totally!